### PR TITLE
Qty box visibility issue in wishlist when product is out of stock

### PR DESCRIPTION
**Pull request by ansari-ziyaurrahman**
### Description (*)

Magento 2.3.x
Magento 2.2.x

Qty box should not be displayed in the wishlist if an added product is out of stock.

### Fixed Issues (if relevant)

1. magento/magento2#<issue_number>: Issue title
2. ...

### Manual testing scenarios (*)

1. Set out of stock product display as a front end from admin configuration by below steps
Stores >> Setting >> Configuration >> Catalog >> Inventory >> Display Out of Stock Products	 set Yes
2. Login as a customer in frontend
3. Add out of stock product into wishlist
4. Review the product in My Wish List section in frontend where qty box is displayed for out of stock product.
Screenshot from 2019 04 17 11 50 01

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-4770](https://jira.corp.magento.com/browse/ENGCOM-4770) has been created to process this Pull Request

**Comment by soleksii** [comment]
:heavy_check_mark: QA Passed 

**Before:**

before

**After:**

after

**Review by kalpmehta — CHANGES_REQUESTED** [review]
Any reason why Qty box should not be displayed if the product is not Saleable? There might be a case when customer want to see how many Qty of the product they added to wishlist, even though the product is out of stock.

**Comment by ansari-ziyaurrahman** [comment]
Hi @kalpmehta,
Thanks for reviewing.
Yes, There might be a case so customer shows Qty box even though the product is out of stock.
But I have not found any condition related to this. When Product is out of stock than "Add to cart" button is not displayed so I have updated the code on based on that condition. When Product is out of stock then also Qty box is in Editable mode, We should remove it or disabled qty box when product is out of stock.
Please let me know your through. If you need to do any change here.
Thank you.

**Comment by kalpmehta** [comment]
@magento-engcom-team give me test instance

**Comment by magento-engcom-team** [comment]
Hi @kalpmehta. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @kalpmehta, here is your [new Magento instance](https://pr-22381.instances.magento-community.engineering).
Admin access: [https://pr-22381.instances.magento-community.engineering/admin](https://pr-22381.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`

**Comment by kalpmehta** [comment]
@ansari-krish Yes, can you please disable the Comment and Qty fields as it's anyway not editable when the product is not Saleable? Thanks

**Comment by ansari-ziyaurrahman** [comment]
> @ansari-krish Yes, can you please disable the Comment and Qty fields as it's anyway not editable when the product is not Saleable? Thanks

Hi @kalpmehta,
I have made changes as per your suggestions, Please review it.
Thanks

**Comment by kalpmehta** [comment]
@magento-engcom-team give me test instance

**Comment by magento-engcom-team** [comment]
Hi @kalpmehta. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @kalpmehta, here is your [new Magento instance](https://pr-22381.instances.magento-community.engineering).
Admin access: [https://pr-22381.instances.magento-community.engineering/admin](https://pr-22381.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`

File: app/code/Magento/Wishlist/view/frontend/templates/item/column/cart.phtml [hunk 1]
```diff
@@ -11,9 +11,7 @@
 /** @var \Magento\Wishlist\Model\Item $item */
 $item = $block->getItem();
 $product = $item->getProduct();
-/** @var \Magento\Wishlist\ViewModel\AllowedQuantity $viewModel */
-$viewModel = $block->getData('allowedQuantityViewModel');
-$allowedQty = $viewModel->setItem($item)->getMinMaxQty();
+$allowedQty = $block->getMinMaxQty();
```
Comment (kalpmehta) [inline app/code/Magento/Wishlist/view/frontend/templates/item/column/cart.phtml]: @ansari-krish Do you have to change this code for this PR?

Comment (ansari-ziyaurrahman) [reply app/code/Magento/Wishlist/view/frontend/templates/item/column/cart.phtml]: No

Comment (kalpmehta) [reply app/code/Magento/Wishlist/view/frontend/templates/item/column/cart.phtml]: Can you please revert it then? thanks!

Comment (ansari-ziyaurrahman) [reply app/code/Magento/Wishlist/view/frontend/templates/item/column/cart.phtml]: ok

Comment (ansari-ziyaurrahman) [reply app/code/Magento/Wishlist/view/frontend/templates/item/column/cart.phtml]: git issue is resolved, Please review now.

**Review by kalpmehta — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @kalpmehta, thank you for the review.
[ENGCOM-4770](https://jira.corp.magento.com/browse/ENGCOM-4770) has been created to process this Pull Request

**Comment by soleksii** [comment]
:heavy_check_mark: QA Passed 

**Result:**

after1

**Pull request merged**